components/dashboard/src/settings/Integrations.tsx

```diff
@@ -199,77 +179,435 @@ function GitProviders() {
+                            </ContextMenu>
+                        </div>
                     </div>
                 </div>
             ))}
         </div>
+        {providers && providers.length > 0 && (
+            <div className="flex justify-start mt-6">
+                <button className="h-full" onClick={() => setModal({ mode: "new" })}>New Git Integration</button>
+            </div>
+        )}
     </div>);
 }
 
+function GitIntegrationModal(props: ({
+    mode: "new",
+} | {
+    mode: "edit",
+    provider: AuthProviderEntry
+}) & {
+    userId: string,
+    onClose?: () => void
+    update?: () => void
+}) {
+
+    const callbackUrl = (host: string) => {
+        const pathname = `/auth/${host}/callback`;
+        return gitpodHostUrl.with({ pathname }).toString();
+    }
+
+    const [type, setType] = useState<string>("GitLab");
+    const [host, setHost] = useState<string>("gitlab.example.com");
+    const [redirectURL, setRedirectURL] = useState<string>(callbackUrl("gitlab.example.com"));
+    const [clientId, setClientId] = useState<string>("");
+    const [clientSecret, setClientSecret] = useState<string>("");
+    const [busy, setBusy] = useState<boolean>(false);
+    const [errorMessage, setErrorMessage] = useState<string | undefined>();
+    const [validationError, setValidationError] = useState<string | undefined>();
+
+    const firstRender = useRef(true);
+
+    useEffect(() => {
+        if (props.mode === "edit") {
+            setType(props.provider.type);
+            setHost(props.provider.host);
+            setClientId(props.provider.oauth.clientId);
+            setClientSecret(props.provider.oauth.clientSecret);
+            setRedirectURL(props.provider.oauth.callBackUrl);
+        }
+    }, []);
+
+    useEffect(() => {
+        if (firstRender.current) {
+            firstRender.current = false
+            return
+        }
+        validate();
+    }, [clientId, clientSecret])
+
+    const close = () => props.onClose && props.onClose();
+    const updateList = () => props.update && props.update();
+
+    const activate = async () => {
+        let entry = (props.mode === "new") ? {
+            host,
+            type,
+            clientId,
+            clientSecret,
+            ownerId: props.userId
+        } as AuthProviderEntry.NewEntry : {
+            id: props.provider.id,
+            ownerId: props.userId,
+            clientId,
+            clientSecret,
+        } as AuthProviderEntry.UpdateEntry
+
+        setBusy(true);
+        setErrorMessage(undefined);
+        try {
+            const newProvider = await getGitpodService().server.updateOwnAuthProvider({ entry });
+
+            // wait two seconds for the changes to be propagated
+            await new Promise(resolve => setTimeout(resolve, 2000));
+
+            updateList();
+
+            // just open the authorization window.
+            /* await */ openAuthorizeWindow({ host: newProvider.host, onSuccess: updateList });
```

Comment:
why is the await in a comment? Do you want to communicate that we explicitly do not await it?
Maybe put it into the comment above.
`// just open the authorization window and don't await`